File: backend/app/schemas/backtest.py
from pydantic import BaseModel, Field


class BacktestRunRequest(BaseModel):
    name: str
    model_keys: list[str]
    start_issue_no: str
    end_issue_no: str
    initial_train_size: int = 100
    candidate_strategy: str = "top_k"


class BacktestMetricRead(BaseModel):
    model_key: str
    backtest_run_id: int | None = None
    total_predictions: int
    avg_red_hits: float
    blue_hit_rate: float
    red_hit_distribution: dict[str, float]
    prize_distribution: dict[str, int]
    random_baseline_diff: dict[str, float] = Field(default_factory=dict)
    ranking_score: float

File: frontend/src/types/backtest.ts
export interface BacktestRunRequest {
  name: string;
  model_keys: string[];
  start_issue_no: string;
  end_issue_no: string;
  initial_train_size: number;
  candidate_strategy: string;
}

export interface BacktestMetric {
  model_key: string;
  backtest_run_id?: number | null;
  total_predictions: number;
  avg_red_hits: number;
  blue_hit_rate: number;
  red_hit_distribution: Record<string, number>;
  prize_distribution: Record<string, number>;
  random_baseline_diff?: Record<string, number>;
  ranking_score: number;
}
